### PR TITLE
feat(cert): Support for extra TLS options

### LINKED ISSUE #2241: eKuiper cannot connect to Azure IoT Hub (probably due to missing RenegotiateOnceAsClient)
**Environment**:

- eKuiper version 1.11
- OS (Alpine Linux v3.16
- Others:

**What happened and what you expected to happen**:
When trying to connect to Azure IoT hub the connection establishes but when publishing the first message the server (on Azure side) closes the connection:

```
time="2023-09-07 01:46:06" level=info msg="The connection to mqtt broker tls://iothub-tfoldi.azure-devices.net:8883 client id rpi-tfoldi-1 established" file="mqtt/mqtt_wrapper.go:86"
time="2023-09-07 01:46:06" level=info msg="Reconnecting to mqtt broker tls://iothub-tfoldi.azure-devices.net:8883 client id rpi-tfoldi-1" file="mqtt/mqtt.go:122"
```

**How to reproduce it (as minimally and precisely as possible)**:

Connect to any Azure service using TLS+MQTT

**Anything else we need to know?**:

I am happy to send a PR to include:

```
Renegotiation: tls.RenegotiateOnceAsClient,
```

at

https://github.com/lf-edge/ekuiper/blob/c4d6c64bfe77eab7ebee821657433f97bd463563/internal/pkg/cert/cert.go#L21

However, you might want to make this a config option to enable users to select if they want to support TLS Renegotation or not. It had some security implications in the past. However, hyper scalers enforce its usage (see: https://github.com/golang/go/issues/5742)

**Question**: do you want to fix or shall I send a PR? If you prefer that I make the PR, shall I make `Renegotiation` a config option (Disable - default, Once, Always).

### PR DESCRIPTION
**Pull request by tfoldi**
Support for extra SSL parameters for `mqtt` and `http` sources. Fixes #2241

### CODE REVIEW

File: internal/pkg/cert/cert.go [hunk 1]
```diff
@@ -9,19 +9,59 @@ import (
...
 func GenerateTLSForClient(
 	Opts TlsConfigurationOptions,
 ) (*tls.Config, error) {
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: Opts.SkipCertVerify,
+		Renegotiation:      getRenegotiationSupport(Opts.RenegotiationSupport),
+		MinVersion:         getTLSMinVersion(Opts.TLSMinVersion),
 	}
 
+	conf.Log.Infof("=> TLS configs: %v.", tlsConfig)
```
Comment (tfoldi) [inline internal/pkg/cert/cert.go:63]: I will remove this line after testing it in a couple of scenarios

File: internal/io/http/client.go [hunk 2]
```diff
@@ -45,17 +45,19 @@ type ClientConf struct {
...
+	Url                  string      `json:"url"`
+	Method               string      `json:"method"`
+	Body                 string      `json:"body"`
+	BodyType             string      `json:"bodyType"`
+	Headers              interface{} `json:"headers"`
+	InsecureSkipVerify   bool        `json:"insecureSkipVerify"`
+	CertificationPath    string      `json:"certificationPath"`
+	PrivateKeyPath       string      `json:"privateKeyPath"`
+	RootCaPath           string      `json:"rootCaPath"`
+	TLSMinVersion        string      `json:"TLSMinVersion"`
```
Comment (ngjaying) [inline internal/io/http/client.go:57]: Can we follow the convention of other properties here. `json:"TLSMinVersion"` ->  `json:"tlsMinVersion"`

File: internal/topo/connection/clients/mqtt/mqtt.go [hunk 3]
```diff
@@ -30,15 +30,17 @@ import (
...
-	InsecureSkipVerify bool   `json:"insecureSkipVerify"`
+	Server               string `json:"server"`
+	PVersion             string `json:"protocolVersion"`
+	ClientId             string `json:"clientid"`
+	Uname                string `json:"username"`
+	Password             string `json:"password"`
+	Certification        string `json:"certificationPath"`
+	PrivateKPath         string `json:"privateKeyPath"`
+	RootCaPath           string `json:"rootCaPath"`
+	TLSMinVersion        string `json:"TLSMinVersion"`
```
Comment (ngjaying) [inline internal/topo/connection/clients/mqtt/mqtt.go:41]: The same here

**Review by ngjaying — CHANGES_REQUESTED** [review]

**Review by ngjaying — CHANGES_REQUESTED** [review]
LGTM. Just need to pass markdown check. Additionally, it'll be better to increase ut coverage to pass the condecov check. Thanks.

**Comment by tfoldi** [comment]
> LGTM. Just need to pass markdown check. Additionally, it'll be better to increase ut coverage to pass the condecov check. Thanks.

The markdown check fails in a file that I did not touch. Actually it is failing due to #2272 committed by @Yisaer  two days ago (see https://github.com/lf-edge/ekuiper/actions/runs/6269824039/job/17026819627).

```
Run markdownlint -c .github/workflows/markdown_config.json ./docs/
  markdownlint -c .github/workflows/markdown_config.json ./docs/
  shell: /usr/bin/bash -e {0}
docs/en_US/configuration/global_configurations.md:[2](https://github.com/lf-edge/ekuiper/actions/runs/6287798441/job/17072623952?pr=2258#step:4:2)[5](https://github.com/lf-edge/ekuiper/actions/runs/6287798441/job/17072623952?pr=2258#step:4:6)9 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
```

Locally, the markdown test passes so "it's not me, it's you" :)

```
❯ markdownlint -c .github/workflows/markdown_config.json docs
❯ echo $?
0
```

For tests, sure, I can add tests for those lines.

**Comment by ngjaying** [comment]
> > LGTM. Just need to pass markdown check. Additionally, it'll be better to increase ut coverage to pass the condecov check. Thanks.
> 
> The markdown check fails in a file that I did not touch. Actually it is failing due to #2272 committed by @Yisaer two days ago (see https://github.com/lf-edge/ekuiper/actions/runs/6269824039/job/17026819627).
> 
> ```
> Run markdownlint -c .github/workflows/markdown_config.json ./docs/
>   markdownlint -c .github/workflows/markdown_config.json ./docs/
>   shell: /usr/bin/bash -e {0}
> docs/en_US/configuration/global_configurations.md:[2](https://github.com/lf-edge/ekuiper/actions/runs/6287798441/job/17072623952?pr=2258#step:4:2)[5](https://github.com/lf-edge/ekuiper/actions/runs/6287798441/job/17072623952?pr=2258#step:4:6)9 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
> ```
> 
> Locally, the markdown test passes so "it's not me, it's you" :)
> 
> ```
> ❯ markdownlint -c .github/workflows/markdown_config.json docs
> ❯ echo $?
> 0
> ```
> 
> For tests, sure, I can add tests for those lines.

Ah, it's weird. The markdown check should have run in that PR. Never mind, we'll fix that. Appreciate your patience.

**Comment by Yisaer** [comment]
plz update the base master branch as we have fixed the markdown lint error

**Review by ngjaying — APPROVED** [review]
LGTM. What a long run, thanks for the patience.

**Pull request merged**